common/nodejs-utils/src/config.rs

```diff
@@ -0,0 +1,338 @@
+use crate::error_handling::ErrorType::UserFacing;
+use crate::error_handling::{
+    error_message, ErrorType, SharedErrorMessageBuilder, SuggestRetryBuild, SuggestSubmitIssue,
+};
+use bullet_stream::style;
+use indoc::formatdoc;
+use std::path::Path;
+use std::str::FromStr;
+use toml_edit::TableLike;
+
+const CONFIG_NAMESPACE: [&str; 5] = ["com", "heroku", "buildpacks", "nodejs", "actions"];
+const PRUNE_DEV_DEPENDENCY_CONFIG: &str = "prune_dev_dependencies";
+
+pub fn read_prune_dev_dependencies_from_project_toml(
+    project_toml_path: &Path,
+) -> Result<Option<bool>, ConfigError> {
+    let project_toml_exists = project_toml_path
+        .try_exists()
+        .map_err(ConfigError::CheckExists)?;
+
+    if !project_toml_exists {
+        return Ok(None);
+    }
+
+    let project_toml = std::fs::read_to_string(project_toml_path)
+        .map_err(ConfigError::ReadProjectToml)
+        .and_then(|contents| {
+            toml_edit::DocumentMut::from_str(&contents).map_err(ConfigError::ParseProjectToml)
+        })?;
+
+    match get_buildpack_namespaced_config(&project_toml)? {
+        Some(namespaced_config) => match namespaced_config.get(PRUNE_DEV_DEPENDENCY_CONFIG) {
+            Some(config) => config.as_bool().map(Some).ok_or(ConfigError::WrongType),
+            None => Ok(None),
+        },
+        None => Ok(None),
+    }
+}
+
+fn get_buildpack_namespaced_config(
+    doc: &toml_edit::DocumentMut,
+) -> Result<Option<&dyn TableLike>, ConfigError> {
+    let mut current_table = doc
+        .as_item()
+        .as_table_like()
+        .ok_or(ConfigError::ExpectedTomlTable)?;
+    for name in CONFIG_NAMESPACE {
+        current_table = match current_table.get(name) {
+            Some(item) => match item.as_table_like() {
+                Some(table) => table,
+                None => return Err(ConfigError::ExpectedTomlTable),
+            },
+            None => return Ok(None),
+        };
+    }
+    Ok(Some(current_table))
+}
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CheckExists(std::io::Error),
+    ReadProjectToml(std::io::Error),
+    ParseProjectToml(toml_edit::TomlError),
+    ExpectedTomlTable,
+    WrongType,
+}
+
+impl From<ConfigError> for SharedErrorMessageBuilder {
+    fn from(value: ConfigError) -> Self {
+        let project_toml = style::value("project.toml");
+        let toml_spec_url = style::url("https://toml.io/en/v1.0.0");
+        let config_table_key = style::value(format!("[{}]", CONFIG_NAMESPACE.to_vec().join(".")));
+        let config_key = style::value(PRUNE_DEV_DEPENDENCY_CONFIG);
+
+        match value {
+            ConfigError::CheckExists(e) => error_message()
+                .error_type(ErrorType::Internal)
+                .header("Failed to check if project.toml exists")
```

Comment:
The subject order implies that the buildpack forgot "failed to check" rather than it checked and the check failed.

```suggestion
                .header("Failed project.toml existence check")
```